### PR TITLE
Documentation on boolean flags is wrong #10517

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit SvenDowideit@home.org.au

closes ##10517

### CODE REVIEW

**Comment by SvenDowideit** [comment]
@fredlf @jamtur01

File: docs/sources/reference/commandline/cli.md [hunk 1]
```diff
@@ -32,19 +32,24 @@ To list the help on any command just execute the command, followed by the `--hel
...
-will set the value to the opposite of the default value.
+Boolean options look like `-d=false`. The value you see in the help text, is the
+default value which is set if you do **not** specify that flag. If you specify
+a Boolean flag without a value, this will set the flag to `true`, irrespective
+of the default value.
+
+For example, if you do call `docker run -d`, that sets the value to `true`, so
+your container **will** run in "detached" mode, in the background.
+
+Options which a defaulted to `true` (i.e., `docker build --rm=true`) can only
```
Comment (duglin) [inline docs/sources/reference/commandline/cli.md]: Perhaps `Options which default to...` ?

**Comment by jamtur01** [comment]
LGTM

**Comment by duglin** [comment]
LGTM thanks for the fix

File: docs/sources/reference/commandline/cli.md [hunk 2]
```diff
@@ -32,19 +32,24 @@ To list the help on any command just execute the command, followed by the `--hel
...
 ### Boolean
 
-Boolean options look like `-d=false`. The value you
-see is the default value which gets set if you do **not** use the
-boolean flag. If you do call `run -d`, that sets the
-opposite boolean value, so in this case, `true`, and
-so `docker run -d` **will** run in "detached" mode,
-in the background. Other boolean options are similar – specifying them
-will set the value to the opposite of the default value.
+Boolean options look like `-d=false`. The value you see in the help text, is the
```
Comment (fredlf) [inline docs/sources/reference/commandline/cli.md]: Omit comma after "text".

File: docs/sources/reference/commandline/cli.md [hunk 3]
```diff
@@ -32,19 +32,24 @@ To list the help on any command just execute the command, followed by the `--hel
...
+default value which is set if you do **not** specify that flag. If you specify
+a Boolean flag without a value, this will set the flag to `true`, irrespective
+of the default value.
+
+For example, if you do call `docker run -d`, that sets the value to `true`, so
```
Comment (fredlf) [inline docs/sources/reference/commandline/cli.md]: Simplify/clarify: For example, running `docker run -d` will set the value to `true`, so your container **will** run in "detached" mode, in the background.

File: docs/sources/reference/commandline/cli.md [hunk 4]
```diff
@@ -32,19 +32,24 @@ To list the help on any command just execute the command, followed by the `--hel
...
+your container **will** run in "detached" mode, in the background.
+
+Options which default to `true` (i.e., `docker build --rm=true`) can only
```
Comment (fredlf) [inline docs/sources/reference/commandline/cli.md]: i.e. should be e.g.

**Comment by fredlf** [comment]
A couple small copy edits. Otherwise, LGTM

**Pull request merged**